legate/core/runtime.py

```diff
@@ -346,6 +358,31 @@ def allocate_field(self, field_size: Any) -> tuple[Region, int, bool]:
         return self._region, field_id, revived
 
 
+def _try_reuse_field(
+    free_fields: Deque[tuple[Region, int, Union[Future, None]]]
+) -> Optional[tuple[Region, int]]:
+    is_ready = False
+    if len(free_fields) == 0:
+        return None
+    count = len(free_fields)
+    while not is_ready and count > 0:
+        field_info = free_fields.popleft()
+        if field_info[2] is not None and field_info[2].is_ready():
+            is_ready = True
+        elif field_info[2] is None:
+            is_ready = True
+        else:
+            free_fields.append(field_info)
+        count -= 1
+    # FIXME: should I return None here instead waiting for future?
```

Comment:
@magnatelee : what would you suggest doing in the case when all fields in the free_fields container have not been deallocated yet: should we call `wait` on the `detach_future`  or return `None`?

Reply:
Let's wait on the future for now. Though any stores with attachments are expected to be small (otherwise, attached allocations couldn't even have been created in a single memory), we don't want to spew stores in the pathological case where the client code keeps creating a store and attaching an allocation to it.